Contrib/ChiralPairs/ChiralDescriptors.py

```diff
@@ -0,0 +1,300 @@
+        legend += '<tspan x="'+str(molSize[0]/2.)+'" y="'+str(molSize[1]+fontSize*2)+'">'+name.split('|')[0]+'</tspan>\n'
+        if len(name.split('|')) > 1:
+            legend += '<tspan x="'+str(molSize[0]/2.)+'" y="'+str(molSize[1]+fontSize*3.5)+'">'+name.split('|')[1]+'</tspan>\n'
+        legend += '</text>\n'
+        blocks[count] = b + legend
+        count+=1
+
+    for i,elem in enumerate(blocks): 
+        row = i//svgsPerRow 
+        col = i%svgsPerRow 
+        elem = rect+elem 
+        blocks[i] = '<g transform="translate(%d,%d)" >%s</g>'%(col*(molSize[0]+molSize[0]/10.0),row*(molSize[1]+labelSizeDist),elem) 
+    res = hdr + '\n'.join(blocks)+ftr 
+    return res 
+
+
+def determineAtomSubstituents(atomID, mol, distanceMatrix, verbose=False):
+    atomPaths = distanceMatrix[atomID]
+    # determine the direct neighbors of the atom
+    neighbors = [n for n,i in enumerate(atomPaths) if i == 1]
+    # store the ids of the neighbors (substituents)
+    sub_dict = defaultdict(list)
+    # track in how many substituents an atom is involved (can happen in rings)
+    sharedNeighbors_dict = defaultdict(int)
+    # determine the max path lenght for each substituent
+    maxShell_dict=defaultdict(int)
+    for n in neighbors:
+        sub_dict[n].append(n)
+        sharedNeighbors_dict[n]+=1
+        maxShell_dict[n]=0
+    # second shell of neighbors
+    mindist=2
+    # max distance from atom
+    maxdist=int(np.max(atomPaths))
+    for d in range(mindist,maxdist+1):
+        if verbose:
+            print("Shell: ",d)
+        newShell = [n for n,i in enumerate(atomPaths) if i == d]
+        for aidx in newShell:
+            if verbose:
+                print("Atom ", aidx," in shell ",d)
+            atom = mol.GetAtomWithIdx(aidx)
+            # find neighbors of the current atom that are part of the substituent already
+            for n in atom.GetNeighbors():
+                nidx = n.GetIdx()
+                for k,v in sub_dict.items():
+                    # is the neighbor in the substituent and is not inthe same shell as the current atom 
+                    # and we haven't added the current atom already then put it in the correct substituent list
+                    if nidx in v and nidx not in newShell and aidx not in v:
+                        sub_dict[k].append(aidx)
+                        sharedNeighbors_dict[aidx]+=1
+                        maxShell_dict[k]=d
+                        if verbose:
+                            print("Atom ",aidx," assigned to ",nidx)
+    if verbose:
+        print(sub_dict)
+        print(sharedNeighbors_dict)
+        
+    return sub_dict, sharedNeighbors_dict, maxShell_dict
+
+def _getSizeOfSubstituents(sub, sharedNeighbors_dict, weighdownShared=True):
+    size=0
+    if weighdownShared:
+        for a in sub:
+            size+=1.0/sharedNeighbors_dict[a]
+        return size
+    return(len(sub))
```

Comment:
I think the construct
```
if condition:
  branchA
  return resultA
return resultB
```
is not as clear as
```
if condition:
  branchA
  return resultA
else:
  return resultB
```

You can actually shorten the calculation of the sum as follows:
```
def _getSizeOfSubstituents(sub, sharedNeighbors_dict, weighdownShared=True):
  if weighdownShared:
    return sum(1.0 / sharedNeighbors_dict[a] for a in sub)
  return len(sub)
```
or clearer:
```
def _getSizeOfSubstituents(sub, sharedNeighbors_dict, weighdownShared=True):
  if weighdownShared:
    return sum(1.0 / sharedNeighbors_dict[a] for a in sub)
  else:
    return len(sub)
```